src/pages/components/DashboardHeader/DashboardHeader.tsx

```diff
@@ -0,0 +1,83 @@
+import { ArrowBackIcon, ChevronRightIcon, CopyIcon } from '@chakra-ui/icons';
+import {
+  Box,
+  Input,
+  InputGroup,
+  InputRightElement,
+  Text,
+  Tooltip,
+  useBreakpointValue,
+  useToast,
+} from '@chakra-ui/react';
+import { useRouter } from 'next/router';
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+interface DashboardHeaderProps {
+  pageTitle: string;
+}
+
+const DashboardHeader: React.FC<DashboardHeaderProps> = function Dashboard({ pageTitle }) {
+  const router = useRouter();
+  const { t } = useTranslation('dashboard');
+  const [apiKey, setApiKey] = useState('');
+  const toast = useToast();
+  const ml = useBreakpointValue({ base: 10, lg: 270 });
+  const pageHeaderDisplay = useBreakpointValue({ base: 'block', lg: 'flex' });
+
+  const handleGoBack = () => {
+    router.back();
+  };
+
+  const handleCopyApiKey = () => {
+    setApiKey('example-api-key');
+    navigator.clipboard.writeText(apiKey);
```

Comment:
instead of hardcoding this `'example-api-key'` you should update this method so that it takes in the API key argument that you can then pass into `setAPIKey`
```suggestion
  const handleCopyApiKey = (userApiKey: string) => {
    setApiKey(userApiKey);
    navigator.clipboard.writeText(userApiKey);
```